src/VirtualClient/VirtualClient.Actions/Blender/BlenderExecutor.cs

```diff
@@ -0,0 +1,229 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO.Abstractions;
+    using System.Runtime.InteropServices;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Logging;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Platform;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+    using VirtualClient.Contracts.Metadata;
+
+    /// <summary>
+    /// Executes the Blender workload.
+    /// </summary>
+    [WindowsCompatible]
+    public class BlenderExecutor : VirtualClientComponent
+    {
+        private const string ResultFilePrefix = "blender_results";
+        private const string BlenderExecutableName = "benchmark-launcher-cli.exe";
+        private IFileSystem fileSystem;
+        private ISystemManagement systemManagement;
+
+        /// <summary>
+        /// Constructor for <see cref="BlenderExecutor"/>
+        /// </summary>
+        /// <param name="dependencies">Provides required dependencies to the component.</param>
+        /// <param name="parameters">Parameters defined in the profile or supplied on the command line.</param>
+        public BlenderExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+            : base(dependencies, parameters)
+        {
+            this.fileSystem = dependencies.GetService<IFileSystem>();
+            this.systemManagement = dependencies.GetService<ISystemManagement>();
+        }
+
+        /// <summary>
+        /// The Scenes that will be run by Blender.
+        /// </summary>
+        public string Scenes
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(BlenderExecutor.Scenes));
+            }
+        }
+
+        /// <summary>
+        /// The Blender Version being used.
+        /// </summary>
+        public string BlenderVersion
```

Comment:
Just curious, why the tool need to care about blender version?

Reply:
The blender benchmark uses blender underneath the hood to render samples. Different blender version can be selected and might affect the rendering speed.